ir_module_compare_wizard/models/ir_module_compare.py

```diff
@@ -0,0 +1,148 @@
+        for record in self:
+            record.module_count = len(record.mapped("import_line_ids.import_module_id")
+                                            
+    def button_open_modules(self):
+        self.ensure_one()
+        modules = self.mapped("import_line_ids.import_module_id")
+        action = self.env["ir.actions.actions"]._for_xml_id(
+            "base.open_module_tree"
+        )
+        action["domain"] = expression.AND(
+            [[("id", "in", modules.ids)], safe_eval(action.get("domain") or "[]")]
+        )
+        action["context"] = dict(self._context, create=False)
+        return action
+
+class IrModuleImportLine(models.Model):
+    _name = "ir.module.import.line"
+    _inherit = "base.import.line"
+    _description = "Wizard lines to import module lines"
+
+    import_id = fields.Many2one(
+        comodel_name="ir.module.import",
+    )
+    import_module_id = fields.Many2one(
+        comodel_name="ir.module.module",
+    )
+    import_module_state = fields.Selection(
+        related="import_module_id.state",
+        store=True,
+    )
+    action = fields.Selection(
+        selection_add=[
+            ("install", "Install"),
+        ],
+        ondelete={"install": "set default"},
+    )
+    module_technical_name = fields.Char(
+        string="Technical Name",
+        states={"done": [("readonly", True)]},
+        required=True,
+    )
+    module_last_version = fields.Char(
+        string="Last Version",
+        states={"done": [("readonly", True)]},
+    )
+    module_website = fields.Char(
+        string="Module Website",
+        states={"done": [("readonly", True)]},
+    )
+    module_author = fields.Char(
+        string="Module Author",
+        states={"done": [("readonly", True)]},
+    )
+    module_notes = fields.Char(
+        string="Module Notes",
+        states={"done": [("readonly", True)]},
+    )
+
+    def _action_validate(self):
+        self.ensure_one()
+        update_values = super()._action_validate()
+        log_infos = []
+        lot = False
+        module, log_info_module = self._check_module()
+        if log_info_module:
+            log_infos.append(log_info_module)
+
+        state = "error" if log_infos else "pass"
+        action = "install" if state != "error" else "nothing"
+        update_values.update(
+            {
+                "import_module_id": module and module.id,              
+                "log_info": "\n".join(log_infos),
+                "state": state,
+                "action": action,
+            }
+        )
+        return update_values
+
+    def _check_module(self):
+        self.ensure_one()
+        log_info = ""
+        if self.import_module_id:
+            return self.import_module_id, log_info
+        module_obj = self.env["ir.module.module"]
+        if self.module_technical_name:
+            search_domain = expression.AND(
+                [[("name", "=", self.module_technical_name)]]
+            )
```

Comment:
NO se necesita este AND, no estas cogiendo un dominio que existe y añadiéndole algo con un AND